tests/FakeItEasy.Tests.Approval/ApiApproval.ApproveApi.approved.txt

```diff
@@ -0,0 +1,600 @@
+        protected void ThrowOnNoTarget() { }
+    }
+    public interface IProxyTargetAccessor
+    {
+        object DynProxyGetTarget();
+        Castle.DynamicProxy.IInterceptor[] GetInterceptors();
+    }
+}
+namespace Castle.DynamicProxy.Generators
+{
+    
+    public class static AttributesToAvoidReplicating
+    {
+        public static void Add(System.Type attribute) { }
+        public static void Add<T>() { }
+        public static bool Contains(System.Type type) { }
+    }
+}
+namespace Castle.DynamicProxy.Internal
+{
+    
+    public abstract class CompositionInvocation : Castle.DynamicProxy.AbstractInvocation
+    {
+        protected object target;
+        protected CompositionInvocation(object target, object proxy, Castle.DynamicProxy.IInterceptor[] interceptors, System.Reflection.MethodInfo proxiedMethod, object[] arguments) { }
+        public override object InvocationTarget { get; }
+        public override System.Reflection.MethodInfo MethodInvocationTarget { get; }
+        public override System.Type TargetType { get; }
+        protected void EnsureValidProxyTarget(object newTarget) { }
+        protected void EnsureValidTarget() { }
+    }
+    public abstract class InheritanceInvocation : Castle.DynamicProxy.AbstractInvocation
+    {
+        protected InheritanceInvocation(System.Type targetType, object proxy, Castle.DynamicProxy.IInterceptor[] interceptors, System.Reflection.MethodInfo proxiedMethod, object[] arguments) { }
+        public override object InvocationTarget { get; }
+        public override System.Reflection.MethodInfo MethodInvocationTarget { get; }
+        public override System.Type TargetType { get; }
+        protected virtual void InvokeMethodOnTarget() { }
+    }
+    public class static TypeUtil
+    {
+        public static System.Reflection.FieldInfo[] GetAllFields(this System.Type type) { }
+        public static System.Type[] GetAllInterfaces(params System.Type[] types) { }
+        public static System.Type[] GetAllInterfaces(this System.Type type) { }
+        public static System.Type GetClosedParameterType(this Castle.DynamicProxy.Generators.Emitters.AbstractTypeEmitter type, System.Type parameter) { }
+        public static System.Type GetTypeOrNull(object target) { }
+        public static bool IsFinalizer(this System.Reflection.MethodInfo methodInfo) { }
+        public static bool IsGetType(this System.Reflection.MethodInfo methodInfo) { }
+        public static bool IsMemberwiseClone(this System.Reflection.MethodInfo methodInfo) { }
+        public static void SetStaticField(this System.Type type, string fieldName, System.Reflection.BindingFlags additionalFlags, object value) { }
+        public static System.Reflection.MemberInfo[] Sort(System.Reflection.MemberInfo[] members) { }
+    }
+}
+namespace FakeItEasy
+{
+    
+    public class static A
+    {
+        [FakeItEasy.Analysis.MustUseReturnValueAttribute("UnusedCallSpecification")]

```

Comment:
@xxxxxxxxxxxxxx to clarify, this PR introduces the use of `PublicApiGenerator`, which generates this output.

`FakeItEasy.Analysis.MustUseReturnValueAttribute` is an `internal` type, so we're surprised that `PublicApiGenerator` thinks it is part of the public API.